api/tls_context.proto

```diff
@@ -39,6 +41,13 @@ message TlsCertificate {
   DataSource password = 3;
   DataSource ocsp_staple = 4;
   repeated DataSource signed_certificate_timestamp = 5;
+
+  // Optional name (e.g. FQDN, UUID, SPKI) by which the certificate can be referred to.
+  // When both: name and kds_config are specified, certificate can be fetched and/or reloaded
+  // via KDS. When name is specified, but kds_config is empty, the certificate will be loaded
+  // from static resources [V2-API-DIFF].
+  string name = 6;
+  ConfigSource kds_config = 7;
```

Comment:
There's a recursive relation here, where a `TlsCertificate` can point to another one delivered by KDS and then the fetched `Secret` can point to another `ConfigSource` etc. Is the intention to support this kind of chaining? Could we instead use a flatter structure, based on `oneof`, to allow folks to say "this cert must be either on the filesystem, inline, or providing by KDS"?

Reply:
The intention was to be able to update inlined TLS certificates via KDS in order to rotate them, but there is indeed a KDS recursion when doing that... `oneof` would indeed be cleaner, but it wouldn't work with the mentioned use case... Good catch, let me think about it.

Reply:
There is something similar here with not allowing the bootstrap clusters to be an EDS cluster. (It makes no sense to have the EDS cluster use EDS). We could potentially just block this in code like we do there.

Reply:
Yeah, looking at EDS, it has the same problem, so I'm not sure if it's worth diverging from it.

I don't have strong opinion here (but slight preference to keep it in line with EDS), so I'll leave it as-is, unless @htuch objects.